source/fab/application.py

```diff
@@ -21,18 +26,42 @@ class Fab(object):
         '.f90': FortranAnalyser,
         '.F90': FortranPreProcessor,
     }
+    _compiler_map: Dict[str, Type[Command]] = {
+        '.f90': FortranCompiler,
+    }
```

Comment:
The fact that we're spawning more of these maps means there may be a missing object which contains them and their attendant use. Something to think about for a future ticket.

Reply:
Yeah I agree, I think we talked about this a little as well before... we might want to rethink the model for this in general; perhaps representing the entire "chain" of classes to process a file from start to finish.  Our current method is maybe a bit awkward for inserting arbitrary extra steps which don't produce any output but do hinge off of the same extension